File: 22_10_2024_Kth_Largest_Sum_in_a_Binary_Tree.java
class Solution {
    public long kthLargestLevelSum(TreeNode root, int k) {
        if (root == null) {
            return 0; // or some indication of invalid input
        }

        List<Long> sums = levelOrder(root);

        Collections.sort(sums, Collections.reverseOrder()); // Sort in descending order

        if (k > sums.size()) {
            return 0; // or some indication of invalid input
        }

        return sums.get(k - 1); // k-th largest means (k-1) in 0-based index
    }

    List<Long> levelOrder(TreeNode root) {
        List<Long> sums = new ArrayList<>();
        List<TreeNode> queue = new ArrayList<>(); // Using List as a queue

        queue.add(root);

        while (!queue.isEmpty()) {
            long levelSum = 0;
            int levelSize = queue.size();

            for (int i = 0; i < levelSize; i++) {
                TreeNode node = queue.remove(0);
                levelSum += node.val;

                if (node.left != null) {
                    queue.add(node.left);
                }
                if (node.right != null) {
                    queue.add(node.right);
                }
            }

            sums.add(levelSum);
        }

        return sums;
    }
}

File: 22_10_2024_Kth_Largest_Sum_in_a_Binary_Tree.py
class Solution(object):
    def kthLargestLevelSum(self, root, k):
        if not root:
            return 0  # Handle empty tree case

        level_sums = []
        queue = deque([root])

        while queue:
            level_sum = 0
            level_size = len(queue)

            for _ in range(level_size):
                node = queue.popleft()
                level_sum += node.val
                
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)

            level_sums.append(level_sum)

        level_sums.sort(reverse=True)

        return level_sums[k - 1] if k <= len(level_sums) else -1
